File: src/game/events/ScoreGameEvent.java
package game.events;

import game.Bomberman;

/**
 * Fires a game event using Strategy Pattern;
 */
public class ScoreGameEvent implements GameEvent {
    @Override
    public void invoke(Object arg) {
        Bomberman.getMatch().getCurrentLevel().onScoreGameEvent((int) arg);
    }
}

File: src/game/domain/events/game/ScoreGameEvent.kt
package game.domain.events.game

import game.JBomb
import game.domain.events.models.GameEvent


class ScoreGameEvent : GameEvent {
    override fun invoke(vararg arg: Any?) {
        JBomb.match.currentLevel.let {
            if (it.info.scoreEnabled)
                it.eventHandler.onScoreGameEventLocal(arg[0] as Int)
        }
    }
}